$OpenBSD$

SPDX-License-Identifier: AGPL-3.0-only

Disambiguate getAddrInfo's result type for newer network/GHC combinations.

Index: src/Simplex/Messaging/Transport/Server.hs
--- src/Simplex/Messaging/Transport/Server.hs.orig
+++ src/Simplex/Messaging/Transport/Server.hs
@@ -235,7 +235,7 @@
   where
     resolve =
       let hints = defaultHints {addrFlags = [AI_PASSIVE], addrSocketType = Stream}
-       in select <$> getAddrInfo (Just hints) host (Just port)
+       in select <$> (getAddrInfo (Just hints) host (Just port) :: IO [AddrInfo])
     select as = fromJust $ family AF_INET6 <|> family AF_INET
       where
         family f = find ((== f) . addrFamily) as
